server/src/http/mod.rs

```diff
@@ -142,10 +142,32 @@ async fn show_console() -> anyhow::Result<Response<Body>> {
 }
 
 async fn handle_request(
+    expected_auth: Option<String>,
     req: Request<Body>,
     sender: mpsc::Sender<Message>,
     enable_console: bool,
 ) -> anyhow::Result<Response<Body>> {
+    if let Some(expected_auth) = expected_auth {
+        let headers = req.headers();
+        match headers.get(hyper::header::AUTHORIZATION) {
+            Some(actual_auth) => {
+                let actual_auth = actual_auth.to_str()?;
```

Comment:
we don't want to return an error here, but a response instead (bad request). We need better error handling, anyhow is terrible for our use case.